src/uu/dd/src/dd.rs

```diff
@@ -71,17 +71,23 @@ impl Input<io::Stdin> {
             iflags,
         };
 
-        // The --skip and --iseek flags are additive. On a stream, they discard bytes.
+        // there is way to seek stdin, so we interpret stdin as a File
+        // to make it seekable
+        #[cfg(any(target_family = "unix", target_family = "wasi"))]
+        let mut seekable_src = unsafe {
+            use std::os::unix::io::{AsRawFd, FromRawFd};
+            std::fs::File::from_raw_fd(std::io::stdin().as_raw_fd())
```

Comment:
(driveby comment, I don't have much context) Unless `seekable_src` is fed to `mem::forget` at some point this likely is a bad idea because `File::drop` will then close the stdin file descriptor. You should either `try_clone` (dup) it or wrap this in a `ManuallyDrop`.

Reply:
That makes sense. Will fix it once I figure out why the current solution works if I use `dd` in command line and does not work if I run tests from `tests/by-utils`.